microservice/src/main/java/com/skytouch/repository/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.skytouch.repository;
+
+import com.skytouch.model.ProductEntity;
+
+import javax.persistence.EntityManager;
+import javax.persistence.StoredProcedureQuery;
+import java.util.List;
+
+public class ProductRepositoryImpl implements ProductRepositoryCustom {
```

Comment:
The class should be default, that is because the service and the implementation are int the same package and the initialization of the service is created for the container (the default scope).

Reply:
Awesome answer. Always keep objects access scope as narrow as you can.